ui/packages/catalog-ui-search/src/main/webapp/component/metacard-overwrite/metacard-overwrite.view.js

```diff
@@ -13,116 +13,306 @@
+`
+
+const OverwriteBack = styled.button`
+  position: absolute;
+  left: 0px;
+  top: 0px;
+  width: ${props => props.theme.minimumButtonSize};
+  height: 100%;
+  text-align: center;
+`
+
+const Confirm = props => (
+  <OverwriteConfirm
+    buttonType={buttonTypeEnum.negative}
+    onClick={props.archive}
+    data-help="This will overwrite the item content. To restore a previous content, you can click on 'File' in the toolbar, and then click 'Restore Archived Items'."
+  >
+    <MainText>Overwrite content</MainText>
+    <SubText>
+      WARNING: This will completely overwrite the current content and metadata.
+    </SubText>
+  </OverwriteConfirm>
+)
+
+const Sending = props => (
+  <OverwriteProgress>
+    <ProgressTextUnder>
+      Uploading File
+      <div>{props.percentage}%</div>
+      <ProgressInfo>
+        If you leave this view, the overwrite will still continue.
+      </ProgressInfo>
+    </ProgressTextUnder>
+    <ProgressText>
+      Uploading File
+      <div>{Math.floor(props.percentage)}%</div>
+      <ProgressInfo>
+        If you leave this view, the overwrite will still continue.
+      </ProgressInfo>
+    </ProgressText>
+    <ProgressBar style={{ width: `${props.percentage}%` }} />
+  </OverwriteProgress>
+)
+
+const Success = props => (
+  <OverwriteSuccess>
+    <OverwriteBack onClick={props.startOver}>
+      <span className="fa fa-chevron-left" />
+    </OverwriteBack>
+    <ResultMessage>{props.message}</ResultMessage>
+  </OverwriteSuccess>
+)
+
+const Error = props => (
+  <OverwriteError>
+    <OverwriteBack onClick={props.startOver}>
+      <span className="fa fa-chevron-left" />
+    </OverwriteBack>
+    <ResultMessage>{props.message}</ResultMessage>
+  </OverwriteError>
+)
+
+const Stages = {
+  Confirm,
+  Sending,
+  Success,
+  Error,
 }
 
-module.exports = Marionette.ItemView.extend({
-  setDefaultModel: function() {
-    this.model = this.selectionInterface.getSelectedResults().first()
-  },
-  template: template,
-  tagName: CustomElements.register('metacard-overwrite'),
-  events: {
-    'click button.overwrite-confirm': 'archive',
-    'click button.overwrite-back': 'startOver',
-  },
-  selectionInterface: store,
-  initialize: function(options) {
-    this.selectionInterface =
-      options.selectionInterface || this.selectionInterface
-    if (!options.model) {
-      this.setDefaultModel()
-    }
-  },
-  onRender: function() {
-    this.setupDropzone()
-  },
-  setupDropzone: function() {
-    this.dropzone = new Dropzone(this.el.querySelector('.overwrite-dropzone'), {
-      url: './internal/catalog/' + this.model.get('metacard').id,
-      maxFilesize: 5000000, //MB
-      method: 'put',
-    })
+const defaultState = {
+  stage: 'Confirm',
+  percentage: 0,
+  message: '',
+}
+
+class MetacardOverwrite extends React.Component {
+  constructor(props) {
+    super(props)
+    this.selectionInterface = props.selectionInterface || store
+    this.state = defaultState
+    this.state.model =
```

Comment:
Is it necessary for the model to be stored on the state? Could you pass the needed model down from the parent component and just reference the model from props?

Reply:
I have moved model out of state